mkosi/backend.py

```diff
@@ -874,17 +543,12 @@ def sort_packages(packages: Iterable[str]) -> list[str]:
     return sorted(packages, key=sort)
 
 
-def scandir_recursive(
-    root: Path,
-    filter: Optional[Callable[[os.DirEntry[str]], T]] = None,
-) -> Iterator[T]:
-    """Recursively walk the tree starting at @root, optionally apply filter, yield non-none values"""
-    queue: Deque[Union[str, Path]] = collections.deque([root])
-
-    while queue:
-        for entry in os.scandir(queue.pop()):
-            pred = filter(entry) if filter is not None else entry
-            if pred is not None:
-                yield cast(T, pred)
-            if entry.is_dir(follow_symlinks=False):
-                queue.append(entry.path)
+def flatten(lists: Iterable[Iterable[T]]) -> list[T]:
+    """Flatten a sequence of sequences into a single list."""
+    return list(itertools.chain.from_iterable(lists))
+
+
+def current_user_uid_gid() -> tuple[int, int]:
+    uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or str(os.getuid()))
```

Comment:
```suggestion
    uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or os.getuid())
```
`int()` is idempotent for ints, so not need to roundtrip this via `str()`